airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/streams.py

```diff
@@ -1,317 +0,0 @@
-    Docs: https://developer.zendesk.com/api-reference/ticketing/ticket-management/incremental_exports
-    """
-
-    # required to support old format as well (only read, but save as new)
-    legacy_cursor_field = "timestamp"
-    cursor_field = "updated_at"
-    filter_param = "start_time"
-
-    def __init__(self, start_date: datetime, **kwargs):
-        super().__init__(**kwargs)
-        self._start_date = pendulum.instance(start_date)
-
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
-        """
-        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
-        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
-        """
-        latest_state = current_stream_state.get(self.cursor_field, current_stream_state.get(self.legacy_cursor_field))
-        new_cursor_value = max(latest_record[self.cursor_field], latest_state or latest_record[self.cursor_field])
-        return {self.cursor_field: new_cursor_value}
-
-    def request_params(self, stream_state=None, **kwargs):
-        """Add incremental parameters"""
-        params = super().request_params(stream_state=stream_state, **kwargs)
-
-        if self.filter_param not in params:
-            # use cursor as filter value only if it is not already a parameter (i.e. we are in the middle of the pagination)
-            stream_state = stream_state or {}
-            state_str = stream_state.get(self.cursor_field, stream_state.get(self.legacy_cursor_field))
-            state = pendulum.parse(state_str) if state_str else self._start_date
-            params[self.filter_param] = max(state, self._start_date).int_timestamp
-
-        return params
-
-    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        """
-        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests. This dict is passed
-        to most other methods in this class to help you form headers, request bodies, query params, etc..
-
-        :param response: the most recent response from the API
-        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
-                If there are no more pages in the result, return None.
-        """
-        next_params = super().next_page_token(response)
-        if not next_params:
-            return None
-
-        current_url = urlparse(response.request.url)
-        current_params = parse_qs(current_url.query)
-
-        # check if cursor value was changed
-        if current_params[self.filter_param] != next_params[self.filter_param]:
```

Comment:
This is indeed weird, but that's because the way the Zendesk Talk API handles pagination in their incremental export endpoints is weird (it's designed to support polling). The old code's logic: if the response's `next_page` url uses the same `start_time` as the current page, then there are no more records, which is correct. I changed the implementation to rely on the response's `count` field instead, to be more in line with [the API documentation](https://developer.zendesk.com/api-reference/voice/talk-api/incremental_exports/):
> Unlike other endpoints that return `null` as the value of `next_page` when reaching the last page of the result set, these endpoints return the URL that should be used for the next export. You should use the count property to determine whether more items are available immediately.